torchtitan/profiling.py

```diff
@@ -49,7 +49,8 @@ def trace_handler(prof):
             # across all ranks. Insert a barrier to make sure all ranks have finished profiling
             # before moving on.
             # TODO: Can we find a cleaner way?
-            torch.distributed.barrier()
+            torch.distributed.barrier(device_ids=[torch.cuda.current_device()])
+            torch.cuda.synchronize()
```

Comment:
If we want to avoid timeout-related issues, I think we should block the CPU until all ranks finish the barrier? (I assume that the timeout starts counting with respect to CPU time)

Reply:
I added this `barrier` trying to do a GPU sync to avoid timeout. However, I later found that the barrier itself can cause timeout if things before are out of sync. So essentially this code doesn't help. I would suggest we remove both.